logstash-core/lib/logstash/api/lib/app/service.rb

```diff
@@ -30,9 +34,10 @@ def update(snapshot)
   def get(key)
     metric_store = @snapshot.metric_store
     if key == :jvm_memory_stats
-      metric_store.get(:root, :jvm, :memory)
+      data = metric_store.get_with_path("jvm/memory")[:jvm][:memory]
+      LogStash::Json.dump(data)
     else
-      { :base => metric_store.get(:root, :base) }
+      metric_store.get_with_path("stats/events")
     end
```

Comment:
The return value of this method is mixed depending on the key we will return a json string or a ruby hash?
I think the serialization should be handled in the representation layer in Sinatra?